### PR TITLE
Overview: Apply design to new bandwidth and site views

### PR DESCRIPTION
**Pull request by chad1008**
Related to: 
- 8399-gh-Automattic/dotcom-forge
- 8400-gh-Automattic/dotcom-forge

## Proposed Changes

Styles the bandwidth usage and site view data on `/overview`, in accordance with the provided design specs. Both of these metrics are displayed for all site, but the CTA link in the bandwidth section will vary depending on the state of the site. Atomic sites get a link taking them to site monitoring, sites with an active plan that are ready to go atomic get a link to activate hosting features, and simple sites that don't have atomic-eligible plans get a link to the upgrade nudge. See design image below.

Each of these new stats will also render a placeholder when necessary (i.e. data isn't available, or is still loading). 

The tree different states of a site:
screen-2024-08-06-at-08 51 50

Placeholders (please ignore the older copy, it's from a previous iteration of this PR):
image

## Why are these changes being made?

- Part of a larger project to bring relevant hosting details to our customers
- see pdKhl6-45r-p2 for full context/discussion

## Testing Instructions

To test this PR, begin by activating the local feature flag:
- Create a .env file in the root of your local calypso repo
- In that file, add `ACTIVE_FEATURE_FLAGS='hosting-overview-refinements'`
- `yarn start`

Then you can test:
- Pull up `/overview` for an active Atomic site. If necessary, visit the site a few times and wait a bit for the visits to be tracked.
- Confirm that underneath the Storage meter, you see Bandwidth data with a "Monitor site performance" link
- Confirm that link takes you to the Monitoring tab
- Back on `/overview` confirm that you also have Site Views data with a "Visit stats" link
- Confirm that link takes you to the site's stats, in Month view
- Create a new site with a Business plan. Do not activate hosting features yet. Do not visit the front end of the site 
- Pull up the new site's `/overview`
- Confirm that underneath the Storage meter, you see Bandwidth listed as "Not available" with a CTA link to activate hosting features
- Confirm that link takes you to the Hosting Features tab where you can see the proper activation button
- Back on `/overview` confirm that you also have Site Views data reporting no "No views yet for this month" and a "Visit stats" link
- Confirm that link takes you to the site's stats, in Month view
- Finally, visit `/overview` for a simple site without an atomic-eligible plan
- Confirm that bandwidth is once again "Not available" and that this time the link takes you to the upgrade nudge

### CODE REVIEW

**Comment by chad1008** [comment]
Updates to this PR now that design is more final and we're moving towards the compact layout for the new stats being added. I have a few small odds and ends on this one to wrap up, and then the addition of the Site Views data (separate PR) should be pretty straight forward, since this one is putting a lot of the design structure in place.

**Comment by chad1008** [comment]
@nuriapenya When you have a moment, I'd love a pair of eyes on the work in progress here

I've applied the designs as they stand so far, but found a few spots where I wasn't sure what our end game design plan was. I'll break them out so they're (hopefully) easy to spot:

1. For sites that haven't gone Atomic yet, I put the following placeholder and link in for now:
Screenshot 2024-08-01 at 21 54 47

I don't remember exactly which iteration of the design I pulled the copy for the link from, but I know it's not the most recent Figma. It looks/feels a little long to me, but I defer to your expertise 🙂

2. I noticed a bit of a delay as some of the data got pulled on page load (particularly the Bandwidth) which caused some jumping in the layout. I threw placeholders (screenshot in description above) to address that. They'll also cover any cases where we just can't access the data for some reason (hopefully never!)

3. I didn't like the look of "Visits: 0 this month", so I replaced that with the some custom text when there are no views to report yet. 
image

I'm not sure If I love that, what do you think? Perhaps something like "Visits: **None yet this month**" would work better?

4. On atomic sites with a staging site, as well as on simple sites, hovering over the storage block pulls up a tooltip:
image

I need to adjust the alignment a bit, it's overlapping the card due to the wrapper I added, but I wanted to check in to see if we were going to keep those more or less the same now that this card isn't dedicated solely to storage.

File: client/hosting-overview/components/plan-card.tsx [hunk 1]
```diff
@@ -172,8 +173,12 @@ const PlanCard: FC = () => {
 	);
 	const isStaging = isStagingSite( site ?? undefined );
 	const isAtomic = useSelector( ( state: AppState ) => isAtomicSite( state, site?.ID ?? 0 ) );
+	const { hasSftpFeature } = useSelector( ( state: AppState ) => ( {
```
Comment (chad1008) [inline client/hosting-overview/components/plan-card.tsx:176]: I'm basing the logic for atomic eligibility on how we handle it for displaying the "activate hosting features" vs the "upgrade" button on the Hosting Settings screen. After testing a bunch this setup seems to work well.

File: client/hosting-overview/components/plan-card.tsx [hunk 2]
```diff
@@ -172,8 +173,12 @@ const PlanCard: FC = () => {
...
+		hasSftpFeature: siteHasFeature( state, site?.ID, FEATURE_SFTP ),
+	} ) );
+	const shouldShowBandwidthStats = isAtomic || ( ! isAtomic && hasSftpFeature );
```
Comment (chad1008) [inline client/hosting-overview/components/plan-card.tsx]: Made an adjustment here (this variable is used to conditionally render the Bandwidth stats down below) because we need to check not just `isAtomic` but also render this for sites that are eligible to go atomic based on their plan.

I'm basing the logic for atomic eligibility on how we handle it for displaying the "activate hosting features" vs the "upgrade" button on the Hosting Settings screen. After testing a bunch this setup seems to work well.

File: client/hosting-overview/components/plan-site-visits.tsx [hunk 3]
```diff
@@ -13,6 +18,10 @@ interface VisitResponse {
 
 export function PlanSiteVisits( { siteId }: PlanSiteVisitsProps ) {
 	const [ visitsNumber, setVisitsNumber ] = useState< number | null >( null );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const canViewStat = useSelector( ( state ) => canCurrentUser( state, siteId, 'publish_posts' ) );
```
Comment (chad1008) [inline client/hosting-overview/components/plan-site-visits.tsx]: Mirroring how the storage bar gates visibility based on user capabilities

File: client/hosting-overview/components/plan-site-visits.tsx [hunk 4]
```diff
@@ -43,32 +52,38 @@ export function PlanSiteVisits( { siteId }: PlanSiteVisitsProps ) {
...
-						moment().format( 'DD/MM/YYYY' )
-					}
-				>
-					{ visitsNumber } views this month
-				</div>
-			</div>
-			Go to Stats page
+		<div className="hosting-overview__plan-site-visits-wrapper">
+			<div className="hosting-overview__plan-site-visits-content">{ getSiteVisitsContent() }</div>
+			<a href={ `/stats/month/${ siteSlug }` }>
```
Comment (chad1008) [inline client/hosting-overview/components/plan-site-visits.tsx]: I've intentionally chosen the month view here, instead of the default daily view, so it aligns with the "this month" messaging presented by the new stat we're rendering

**Comment by chad1008** [comment]
Updated design applied, we should be good for review! Thank you again for all of the help on making this look right @nuriapenya!

**Comment by nuriapenya** [comment]
> Updated design applied, we should be good for review! Thank you again for all of the help on making this look right @nuriapenya!

Hey @chad1008 👋🏽 for some reason I can't test this PR. The new sections doesn't appear for me (probably I'm doing something wrong). I opened the Calypso test link on an active Atomic site, but I can't see anything:

screen 2024-08-07 at 11 08 42

**Comment by chad1008** [comment]
oof, I'm sorry @nuriapenya, I left a critical step out of the testing instructions. I'll update them now.

**Comment by fredrikekelund** [comment]
Per [these docs](https://wpcalypso.wordpress.com/devdocs/config/README.md#testing-feature-flags-via-urls), feature flags can be activated by passing a query parameter in the URL. In this specific case, `?flags=hosting-overview-refinements`.

So instead of creating a `.env` file (which requires running Calypso locally), you can use the Calypso preview link. Navigate to `/overview/:site_slug`, add `?flags=hosting-overview-refinements` to the URL and hit <kbd>Enter</kbd>

That might help your testing, @nuriapenya.

File: client/hosting/overview/components/plan-bandwidth.tsx [hunk 5]
```diff
@@ -33,35 +60,85 @@ export function PlanBandwidth( { siteId }: PlanBandwidthProps ) {
 		metric: 'response_bytes_persec',
 	} );
 
-	if ( ! data || ! selectedSiteDomain ) {
+	useEffect( () => {
+		if ( data ) {
+			setBandwidthData( data );
+		}
+	}, [ data ] );
```
Comment (fredrikekelund) [inline client/hosting/overview/components/plan-bandwidth.tsx]: This looks redundant to me. I see that you added this in 5aa923aa5e6134fdb27429a436277804fe3ac1c6 and mentioned "avoiding placeholder flashes". Could you expand on which problem you were solving?

File: client/hosting/overview/components/plan-bandwidth.tsx [hunk 6]
```diff
@@ -22,9 +36,22 @@ const getCurrentMonthRangeTimestamps = () => {
...
 export function PlanBandwidth( { siteId }: PlanBandwidthProps ) {
+	const [ bandwidthData, setBandwidthData ] = useState< SiteMetricsAPIResponse >();
 	const selectedSiteData = useSelector( getSelectedSite );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const isAtomic = useSelector( ( state ) => isAtomicSite( state, siteId ) );
+	const canViewStat = useSelector( ( state ) => canCurrentUser( state, siteId, 'publish_posts' ) );
+	const { isSiteAtomic, hasSftpFeature } = useSelector( ( state ) => ( {
+		isSiteAtomic: isSiteWpcomAtomic( state, siteId as number ),
+		hasSftpFeature: siteHasFeature( state, siteId, FEATURE_SFTP ),
+	} ) );
```
Comment (fredrikekelund) [inline client/hosting/overview/components/plan-bandwidth.tsx]:
```suggestion
	const isSiteAtomic = useSelector( ( state ) => isSiteWpcomAtomic( state, siteId as number ) );
	const hasSftpFeature = useSelector( ( state ) => siteHasFeature( state, siteId, FEATURE_SFTP ) );
```
Destructuring seems redundant here and it's unclear why we treat these variables differently from the ones above. I would just use two separate `useSelector` calls.

File: client/hosting/overview/components/plan-bandwidth.tsx [hunk 7]
```diff
@@ -22,9 +36,22 @@ const getCurrentMonthRangeTimestamps = () => {
 };
 
 export function PlanBandwidth( { siteId }: PlanBandwidthProps ) {
+	const [ bandwidthData, setBandwidthData ] = useState< SiteMetricsAPIResponse >();
 	const selectedSiteData = useSelector( getSelectedSite );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const isAtomic = useSelector( ( state ) => isAtomicSite( state, siteId ) );
+	const canViewStat = useSelector( ( state ) => canCurrentUser( state, siteId, 'publish_posts' ) );
+	const { isSiteAtomic, hasSftpFeature } = useSelector( ( state ) => ( {
+		isSiteAtomic: isSiteWpcomAtomic( state, siteId as number ),
```
Comment (fredrikekelund) [inline client/hosting/overview/components/plan-bandwidth.tsx]: What's the difference between `isAtomicSite` and `isSiteWpcomAtomic`? AFAICT, they're retrieved from different API endpoints, but I assume they indicate the same thing. 

Is there a difference?

**Review by fredrikekelund — COMMENTED** [review]
I left some quick initial comments, but I haven't tested it properly yet

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 5 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: For the bandwidth data specifically, I found that subsequent re-renders would cause the placeholder to reappear after the initial render as the data was being reloaded periodically. Based on the timing of these from my observation, I believe it was caused by other components on the page triggering a re-render, at which point our component would retrieve the bandwidth data again. Whenever that happened, the bandwidth data wouldn't be present and the placeholder would reappear. Storing the value in state and updating it only when there's an actual result keeps the data available and visible on the page.

I experimented with a few different solutions, including a simpler `useMemo` but this approach seemed to work best... but I'm happy to update or look again if you have suggestions 🙂

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 6 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: 👍 Thanks! Looking back, these wound up being treated separately because I added this logic after the initial variables, and was emulating [the condition used to determine whether or not to display an activation button for hosting features](https://github.com/Automattic/wp-calypso/blob/4d73c032eb95cbd50ebae7a223a74e4b895560a5/client/hosting/hosting-features/components/hosting-features.tsx#L47-L48). Since we're matching that state (ready to go atomic) I wanted to use consistent logic, and I wound up carrying the destructuring over from there. You're right though, it's more consistent and makes more sense with your suggestion.

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 7 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: You're right, they do indicate the same thing, having both was an oversight when incorporating a later change. I'm pushing a change now to clean this up, thanks for spotting it.

**Comment by chad1008** [comment]
@nuriapenya How do you feel about an alternate take one the titles for these new metrics? Something about the current design was bugging me so I wanted to test another option:

image

It does make the title format different from Storage by losing the all caps, but maybe that's okay? Since we don't have icons for these, and they're not full width, the overall layout is already fairly different. If we went this route, maybe also adding "this month" to the end of the bandwidth measurement would also make sense?

I'm not going to push any changes to the design, but was curious what you thought 🙂

**Comment by nuriapenya** [comment]
> Something about the current design was bugging me

@chad1008 yeah, I had the same feeling. As we add new sections, I would like to keep as much consistency as possible, even if that means tweaking a bit of the current Storage design. I've been playing around with the design, and I came up with this solution:

(Figma: R8yjT7mjG0Uw8h2GPyIY6R-fi-4041_65153)

screen 2024-08-08 at 08 08 38

* Removing icon from Storage
* Keep labels All caps
* Highlight the actual data in bold
* All links aligned on the left, and in blue
* I would separate the Unlimited from the label to make it a bit more outstanding, as it's part of our marketing strategy.

And I also tweaked a bit the spacing & padding, to make it more compact and visually well balanced.

screen 2024-08-08 at 08 12 12

I know this adds a bit more work, but I really think it's worth spending some time on nailing this area, which is the core part of the project 🙏🏽 let me know what you think.

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 5 above)
Comment (fredrikekelund) [reply client/hosting/overview/components/plan-bandwidth.tsx]: I think the issue is that because the `endInSeconds` value returned from `getCurrentMonthRangeTimestamps` isn't stable, we don't get a stable query key in `useSiteMetricsQuery`. I.e., `endInSeconds` is always the current time instead of a consistent time, like 23:59:59, for example.

If we had a stable query key, the data wouldn't be refetched on every render. 

I propose removing this logic and making `endInSeconds` stable instead.

Comment (fredrikekelund) [reply client/hosting/overview/components/plan-bandwidth.tsx]: Adjusting `endInSeconds` so it always returns the last second of the current hour seems like a good idea when I think about it. That way, the data still updates throughout the day, just not on every render

File: client/hosting/overview/components/style.scss [hunk 8]
```diff
@@ -212,27 +205,72 @@ $blueberry-color: #3858e9;
...
+
 .hosting-overview__plan-storage-title-wrapper,
 .hosting-overview__plan-storage-title {
 	display: flex;
 	align-items: center;
 }
 
-.hosting-overview__plan-storage-title {
+.hosting-overview__plan-storage-title,
+.hosting-overview__plan-bandwidth-title span,
```
Comment (nightnei) [inline client/hosting/overview/components/style.scss]: Based on my experience - it's bad practice to use a selector as tag-name, since tomorrow somebody adds one more `span` inside of `.hosting-overview__plan-bandwidth-title`, and it will make maintenance more difficult.
Also it's easier to read selector names when we have classname, instead of `span`, since at a glance it's totally not clear what is the `span`.

File: client/hosting/overview/components/style.scss [hunk 9]
```diff
@@ -212,27 +205,72 @@ $blueberry-color: #3858e9;
...
+
+.hosting-overview__plan-storage,
+.hosting-overview__plan-bandwidth,
+.hosting-overview__plan-site-visits {
+	display: flex;
+	flex-direction: column;
+	border-radius: 4px;
+	border: 1px solid var(--studio-gray-0);
+	background: var(--studio-gray-0);
+	padding: var(--grid-unit-30, 24px);
```
Comment (nightnei) [inline client/hosting/overview/components/style.scss]: I don't understand this approach swhich is used all around Calypso, could you please explain - what is the reason to use `--grid-unit-30` and as a fallback `24px`? 
Additionally fallback 24px for the 30px looks confusing.

Also, I don't see declaration of `--grid-unit-30`.

For me looks like some magic 😅 Maybe I am missing something

File: client/hosting/overview/components/plan-card.tsx [hunk 10]
```diff
@@ -251,30 +248,34 @@ const PlanCard: FC = () => {
...
+												plain
+												href={ `/add-ons/${ site?.slug }` }
+											>
+												{ translate( 'Need more storage?' ) }
+											</Button>
+										</div>
+									) }
+								</PlanStorage>
+								{ config.isEnabled( 'hosting-overview-refinements' ) && site && (
+									<>
```
Comment (nightnei) [inline client/hosting/overview/components/plan-card.tsx]: We don't need it
```suggestion
```

File: client/hosting/overview/components/plan-card.tsx [hunk 11]
```diff
@@ -251,30 +248,34 @@ const PlanCard: FC = () => {
...
+										<div className="hosting-overview__site-metrics-footer">
+											<PlanBandwidth siteId={ site.ID } />
+											<PlanSiteVisits siteId={ site.ID } />
+										</div>
+									</>
```
Comment (nightnei) [inline client/hosting/overview/components/plan-card.tsx]:
```suggestion
```

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 5 above)
Comment (nightnei) [reply client/hosting/overview/components/plan-bandwidth.tsx]: We can consider updating the hook with the opportunity to provide `enabled: false` (or something else, don't remember exactly how to prevent refetching). AFAIU it fixes the initial issues. 
Also, I agree with Fredrik - `useEffect` here it's just workaround, which looks not nice.

File: client/hosting/overview/components/plan-bandwidth.tsx [hunk 12]
```diff
@@ -22,9 +35,19 @@ const getCurrentMonthRangeTimestamps = () => {
...
+
+	const isEligibleForAtomic = ! isAtomic && hasSftpFeature;
+
 	const selectedSiteDomain = selectedSiteData?.domain;
 
+	const translate = useTranslate();
+
 	const { startInSeconds, endInSeconds } = getCurrentMonthRangeTimestamps();
 
 	const { data } = useSiteMetricsQuery( siteId, {
```
Comment (nightnei) [inline client/hosting/overview/components/plan-bandwidth.tsx]: Before we had check for `isAtomic`, to avoid redundant requests to the server. It doesn't make sense to overload server when it obviously returns error.
Let's disable this request for simple websites

**Review by nightnei — CHANGES_REQUESTED** [review]
Works well 👍 But there are a few comments to address

Screenshot 2024-08-08 at 13 35 49
Screenshot 2024-08-08 at 13 43 03

**Comment by chad1008** [comment]
> I know this adds a bit more work, but I really think it's worth spending some time on nailing this area, which is the core part of the project 🙏🏽 let me know what you think.

I think separating the "Unlimited" from the label makes things look a lot better, while maintaining the consistency your after between the different sections. I'll incorporate the updated design while working addressing other PR feedback 🙂

File: client/hosting/overview/components/style.scss (hunk 9 above)
Comment (chad1008) [reply client/hosting/overview/components/style.scss]: You're not wrong, it's kind of magical.

The `grid-unit` system feels familiar to me because I used it when working on Gutenberg... and when I saw it (with added fallbacks) in Calypso, I just kind of adapted and incorporated the fallbacks in the interest of consistency.

The grid units themselves are a [spacing system used in Core](https://github.com/WordPress/gutenberg/blob/5d408ce8849e6183cb39f48b071400b04a1cde78/packages/base-styles/_variables.scss#L25-L40). It's based on an 8px unit, named `--grid-unit-10`. From there, every ten `grid-units` is another 8px. That's why `grid-unit-30` equates to `24px`, though I agree those numbers feel awkward side by side. Seeing the math in the actual declarations I've linked above helped me wrap my brain around it when I started using it, and there's more context in [the proposal post](https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/) if you're curious 🙂 

From what I understand, I think the reason we're seeing `grid-unit`+`fallback` pattern in Calypso is to prepare for a potential future where we incorporate that system here. I'm not sure how organized of an effort that is, or if it's more just individual developers proactively future-proofing. I know there were [discussions about incorporating this within woo a while back](10045-gh-Automattic/woocommerce.com/files#diff-8a2ef6dca3124e9c64d765dd2e6663fd2f1543279068ae9ea9e3e1399aea3fee), but I wasn't able to find similar conversatons about calypso. Granted, the words "grid" "unit" and "Calypso" are utterly _horrible_ ingredients for a useful search term, so it's likely there's discussions/context out there I didn't come across in my brief search.

So, tl;dr: the `px` fallbacks are doing the real work here, but it looks like we're at least partially preparing for a day when a universal spacing system could be implemented, and putting the variables in place ahead of time.

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 5 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: Thank you both for the input. The updates to `endInSeconds` did indeed appear to be the culprit. I've adjusted  `endInSeconds` as @fredrikekelund suggested to return the final second of the current hour and we look to be in good shape on the re-fetching side of things now.

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 12 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: I don't recall having this in an `isAtomic` check (React would have hated a conditional hook call like that) but you make an excellent point about not requesting data we know isn't available.

I've encapsualted the logic for retrieving atomic site bandwidth data in a separate sub-component, so `useSiteMetricsQuery` only fires when we actually need it to, but without violating the rules of hooks. This also introduced a small change where if we don't have a valid `selectedSiteDomain` for some reason, we'll get the same "Not available" output a simple site would have, instead of a perpetual placeholder.

**Comment by chad1008** [comment]
Current feedback and design updates should all be addressed. I've also made sure the two new sections will stack vertically on small screens. I the smallest of our common breakpoints because it renders pretty well until we get to the tiniest of devices.

Thank you so much to everyone... @fredrikekelund, @nightnei, @nuriapenya for the multiple rounds of feedback to get this one just right!

File: client/hosting/overview/components/plan-bandwidth.tsx [hunk 13]
```diff
@@ -1,10 +1,19 @@
...
+import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 
 interface PlanBandwidthProps {
+	children?: React.ReactNode;
 	siteId: number;
+	className?: string;
 }
```
Comment (fredrikekelund) [inline client/hosting/overview/components/plan-bandwidth.tsx]:
```suggestion
type PlanBandwidthProps = React.PropsWithChildren< {
	siteId: number;
} >;
```
Nit, but I typically advise using `PropsWithChildren` instead of manually typing the `children` prop. Also, it looks like we aren't using the `className` prop, so let's remove that.

Lastly, I'd advise moving this definition closer to where it's used (so just above `PlanBandwidth`)

File: client/hosting/overview/components/style.scss (hunk 9 above)
Comment (fredrikekelund) [reply client/hosting/overview/components/style.scss]: Outside `client/hosting/overview/components/style.scss`, I see only 2 instances of `--grid-unit-*` CSS variables being used in Calypso, which makes me question if we should really add them here. Is that correct, or am I missing something, @chad1008?

I absolutely see the appeal in having a system for spacing, but the low usage makes me think that the effort to implement this in Calypso might not be organized or it might have stalled.

**Review by fredrikekelund — COMMENTED** [review]
This is looking good and tests well 👍 I'll leave the final approval to @nightnei, but I also left some small comments

File: client/hosting/overview/components/style.scss (hunk 9 above)
Comment (nightnei) [reply client/hosting/overview/components/style.scss]: I have similar thoughts as Fredrik.
Additionally - `grid-unit`s doesn't work ATM, so there is possibility to make mistake as something will be broken as soon as they are integrated to Calypso. So anyway, it will be necessary to double-check all of them. And long story short, I would say - YAGNI.

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 12 above)
Comment (nightnei) [reply client/hosting/overview/components/plan-bandwidth.tsx]: The condition was [here](https://github.com/Automattic/wp-calypso/pull/93032/files#diff-cf17951d9865e6a59dff8cbc5181302e1e48230b11f75b69fc09801f45a6de2dL279). It's removed in this PR.

And I meant not conditional query, but disabling running the query in case of atomic.
https://tanstack.com/query/v4/docs/framework/react/guides/disabling-queries#:~:text=If%20you%20ever%20want%20to,'success'%20or%20isSuccess%20state.

**Review by nightnei — APPROVED** [review]
In general it's LGTM, so let's merge it. And if you have time today or Monday - let's consider these two points as polishments, ok?
https://github.com/Automattic/wp-calypso/pull/93032#discussion_r1711486583
https://github.com/Automattic/wp-calypso/pull/93032#discussion_r1711471376

**Comment by chad1008** [comment]
> Outside client/hosting/overview/components/style.scss, I see only 2 instances of --grid-unit-* CSS variables being used in Calypso, which makes me question if we should really add them here. Is that correct, or am I missing something, @chad1008? I absolutely see the appeal in having a system for spacing, but the low usage makes me think that the effort to implement this in Calypso might not be organized or it might have stalled.

Good point. I was happy to see them in use, but didn't think to check how wide spread it was. I do only see two instances of `--grid-unit*` CSS variables with fallbacks. Using them as Sass variables ($grid-unit-*) appears a little more widespread, popping up in 15 files across Calypso. Still not massive usage though, so perhaps you're right and we should omit them here. I have reached out to @xavier-lc, who authored the PR who introduced grid-units when setting up Hosting Overview, to see if they have any context to add on `grid-unit` in Calypso (is it an organized effort or an individual dev preference)

File: client/hosting/overview/components/plan-bandwidth.tsx (hunk 12 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-bandwidth.tsx]: Oh, I see... you meant when the entire `PlanBandwidth` component was behind an `isAtomic` condition, not that the `useSiteMetricsQuery` was somehow being handled conditionally. That makes more sense, thank you for clarifying!

I'll look into utilizing `enabled: false` as a potential followup!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16467105

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @chad1008 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16467105).
